### DISCUSSION TITLE
Asset stream error with S3 configuration in Pimcore X

### DISCUSSION THREAD

**Discussion by Birger78** [pimcore/pimcore#9293] [Q&A / Support] [answered]
Hi,

I'm trying to use the AWS S3 file storage together with Pimcore X.

The storage is configured in "flysytem.yaml" which is then imported by the "config.yaml" file und contains the setup following https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/File_Storage_Setup.html

I required the composer package "league/flysystem-aws-s3-v3" in my project.

When I try to open an asset in the Pimcore X asset view, an exception is thrown as response to "/admin/asset/get-image-thumbnail?id=ASSET_ID":

**fpassthru(): Argument #1 ($stream) must be of type resource, null given**

**in vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/Asset/AssetController.php (line 1322)** 

```
$stream = $thumbnail->getStream();
        $response = new StreamedResponse(function () use ($stream) {
            fpassthru($stream);
        }, 200, [
            'Content-Type' => $thumbnail->getMimeType(),
            'Access-Control-Allow-Origin', '*',
        ]);
```

Did I maybe miss anything setting up the S3 storage connection?

Any hints and thoughts are very welcome ;)

Thanks,
Birger

**Comment by ghost** [comment, marked as answer]
hi, 

with the following configuration i managed to upload assets to my S3 bucket.

my directory structure looks like:
![image](https://user-images.githubusercontent.com/68632142/120337077-48d9b200-c2f3-11eb-8298-e0cbaf38766b.png)

i changed in .env:
APP_ENV=prod

i created  an .env.local file with aws credentials
AWS_S3_BUCKET=example-bucket
AWS_KEY=A1111111
AWS_SECRET= byhyhyhyhyhyyyyyyy

in config/packages/prod i created flysystem.yaml

![image](https://user-images.githubusercontent.com/68632142/120338674-bfc37a80-c2f4-11eb-9905-cd77a2698519.png)

in config/local i created database.yaml

![image](https://user-images.githubusercontent.com/68632142/120339245-3c565900-c2f5-11eb-955c-d6ad92308b23.png)

finally i cleared var/cache/prod

**Reply by Birger78** [reply]
Hi @Silverado2791 ,
thank you very much for your answer that gave me the solution - the "prefix"es were not set in my flysystem.yaml!
Now I'm waiting for Pimcore X 10.0.2 and the fix of the bug https://github.com/pimcore/pimcore/pull/9234 to be able to download S3 assets from Pimcore backend again...
Cheers
Birger

**Comment by ghost** [comment]
Hi Birger,

 

Thank you for your answer. Well there are stil some strange behaviours and I hope that these issues will be fixed soon. 

When I move an asset folder with its content in the directory tree then the full path of the folder and the assets should be updated respectively

 

But the actual behaviour is different. 

If I move an asset folder with  its content in the directory tree then the full path of the folder is being updated but the full path of the the assets inside the folder does not change.

 

Example:

Copy the TESTFOLDER with the jpg file  from /ORG/HRST/ANW/ROL/TESTFOLDER/G55064_115_V4.jpg  to  the new destination /ORG/HRST/TESTFOLDER/G55064_115_V4.jpg

 

Original path: /ORG/HRST/ANW/ROL/TESTFOLDER/G55064_115_V4.jpg

 

Expected path: /ORG/HRST/TESTFOLDER/G55064_115_V4.jpg

 

Actual path: /ORG/HRST/ANW/ROL/TESTFOLDER/G55064_115_V4.jpg

 

 

Steps to reproduce:

 

1. Right click on specific asset folder and select “Cut”

2. Right click on destination folder and select “Paste cut-out element”

3. Left click on specific asset in the moved folder and from the menu “copy full path to clipboard”

Can you reproduce this behaviour on your system?



Cheers

Heiko
